lib/txmgr/txmgr.go

```diff
@@ -493,7 +499,9 @@ func (m *SimpleTxManager) queryReceipt(ctx context.Context, txHash common.Hash,
 	defer cancel()
 	receipt, err := m.backend.TransactionReceipt(ctx, txHash)
 	if err != nil {
-		if errors.Is(err, ethereum.NotFound) || strings.Contains(err.Error(), ethereum.NotFound.Error()) {
+		if strings.Contains(err.Error(), "transaction indexing is in progress") {
```

Comment:
Not [a Go-idiomatic way of working with errors](https://go.dev/blog/go1.13-errors). We should always use `errors.Is` and `errors.As` to check if an error is of particular type.

Reply:
I just realized that this error is inside of the `internal` ethereum package and we can't call `errors.Is` on it 🤦 .
I'll raise an issue on the Ethereum repo.

Reply:
I submit [an issue upstream](https://github.com/ethereum/go-ethereum/issues/29063) to make that error public. The docs say that there is no receipts for pending transactions. Alternatively, we can try to get the transaction by hash and verify that it's not pending and only then get the receipt without any string comparison. Otherwise, if the error text changes, our code will stop working as expected with a silent bug.